src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -560,9 +560,12 @@ trait Infer extends Checkable {
       // explicitly anywhere amongst the formal, argument, result, or expected type.
       // ...or lower bound of a type param, since they're asking for it.
       def canWarnAboutAny = {
-        val loBounds = tparams map (_.info.lowerBound)
-        def containsAny(t: Type) = (t contains AnyClass) || (t contains AnyValClass) || (t contains ObjectClass)
-        val hasAny = pt :: restpe :: formals ::: argtpes ::: loBounds exists (_.dealiasWidenChain exists containsAny)
+        val coll = new ContainsAnyCollector(List(AnyClass, AnyValClass, ObjectClass))
```

Comment:
List(AnyClass, AnyValClass, ObjectClass) is a constant expression and can be moved out of the function.

Reply:
I was going to joke that despite the title, it still has a list concatenation. I'm sorry I missed my shot. One could also define a dedicated `ContainsAnyAnyCollector`.